apps/preview/next/pages/showcases/OrderSummary/OrderSummary.tsx

```diff
@@ -23,68 +26,138 @@ export default function OrderSummary() {
-            <p>Estimated Sales Tax</p>
-          </div>
-          <div className="flex flex-col text-right">
-            <p>{formatPrice(itemsSubtotal())}</p>
-            <p className="typography-text-xs text-neutral-500">{formatPrice(orderDetails.originalPrice)}</p>
-            <p className="typography-text-xs text-secondary-700">{formatPrice(orderDetails.savings)}</p>
-            <p className="my-2">{formatPrice(orderDetails.delivery)}</p>
-            <p>{formatPrice(orderDetails.tax)}</p>
-          </div>
+    <div>
+      <div className="md:shadow-lg md:rounded-md md:border md:border-neutral-100">
+        <div className="flex justify-between items-end bg-neutral-100 md:bg-transparent py-2 px-4 md:px-6 md:pt-6 md:pb-4">
+          <p className="typography-headline-4 font-bold md:typography-headline-3">Order summary</p>
+          <p className="typography-text-base font-semibold">(Items: {orderDetails.items})</p>
         </div>
-        {!!promoCode && (
-          <div className="flex items-center py-4 border-t border-neutral-200">
-            <p>PromoCode</p>
-            <SfButton size="sm" variant="tertiary" className="ml-auto mr-2" onClick={removePromoCode}>
-              Remove
+        <div className="px-4 pb-4 mt-3 md:px-6 md:pb-6 md:mt-0">
+          <div className="flex justify-between typography-text-base pb-4">
+            <div className="flex flex-col grow pr-2">
+              <p>Items Subtotal</p>
+              <p className="ml-auto typography-text-xs text-neutral-500">Original Price</p>
+              <p className="ml-auto typography-text-xs text-secondary-700">Savings</p>
+              <p className="my-2">Delivery</p>
+              <p>Estimated Sales Tax</p>
+            </div>
+            <div className="flex flex-col text-right">
+              <p>{formatPrice(itemsSubtotal())}</p>
+              <p className="typography-text-xs text-neutral-500">{formatPrice(orderDetails.originalPrice)}</p>
+              <p className="typography-text-xs text-secondary-700">{formatPrice(orderDetails.savings)}</p>
+              <p className="my-2">{formatPrice(orderDetails.delivery)}</p>
+              <p>{formatPrice(orderDetails.tax)}</p>
+            </div>
+          </div>
+          {!!promoCode && (
+            <div className="flex items-center py-4 border-t border-neutral-200">
+              <p>PromoCode</p>
+              <SfButton size="sm" variant="tertiary" className="ml-auto mr-2" onClick={removePromoCode}>
+                Remove
+              </SfButton>
+              <p>{formatPrice(promoCode)}</p>
+            </div>
+          )}
+          <div className="flex gap-x-2 py-4 border-y border-neutral-200 mb-4">
+            <SfInput
+              value={inputValue}
+              placeholder="Enter promo code"
+              wrapperClassName="grow"
+              onChange={(event) => setInputValue(event.target.value)}
+            />
+            <SfButton variant="secondary" onClick={checkPromoCode}>
```

Comment:
I think this should be a `<form>` with submit button. IMO submit handler should contain validator to check given promo code, and then respond with a proper alert.

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2627/commits/3f5d1a8e6e56c7addd34fcbdc6d9c5233ceeec23